src/main/java/net/tropicraft/core/common/entity/ai/EntityAIPartyTime.java

```diff
@@ -0,0 +1,365 @@
+        {
+            if (!isTooClose()) {
+                return true;
+            } else {
+                return false;
+            }
+
+        } else {
+            if (entityObj.world.rand.nextInt(60) == 0) {
+                return false;
+            } else {
+                return true;
+            }
+        }
+    }
+
+    @Override
+    public void updateTask() {
+        super.updateTask();
+
+        boolean isClose = false;
+
+        BlockPos blockposGoal = null;
+        if (this.entityObj.listPosDrums.size() > assignedDrumIndex) {
+            blockposGoal = entityObj.listPosDrums.get(assignedDrumIndex);
+        }
+
+        if (entityObj.world.getTotalWorldTime() % 200 == 0){
+            if (this.entityObj.listPosDrums.size() > 0) {
+                assignedDrumIndex = entityObj.world.rand.nextInt(entityObj.listPosDrums.size());
+            }
+            //if (wasClose) {
+                bangDrum = entityObj.world.rand.nextBoolean();
+            //}
+        }
+
+        if (blockposGoal == null) {
+            resetTask();
+            return;
+        }
+
+        //prevent walking onto source
+        double dist = entityObj.getPositionVector().distanceTo(new Vec3d(blockposGoal.getX(), blockposGoal.getY(), blockposGoal.getZ()));
+        if (dist < 8D) {
+            wasClose = true;
+        }
+        if (dist < 3D && entityObj.onGround) {
+            isClose = true;
+            entityObj.getNavigator().clearPathEntity();
+            if (!bangDrum) {
+                //entityObj.setSitting(true);
+                entityObj.setDancing(true);
+                this.entityObj.getJumpHelper().setJumping();
+                this.entityObj.rotationYaw = entityObj.world.rand.nextInt(360);
+            } else {
+                entityObj.setDancing(false);
+                if (true || lookUpdateTimer <= 0) {
+
+                    entityObj.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, null);
+
+                    //keep for testing, was neat sounding
+                    int amp = 1;//entityObj.world.rand.nextInt(10) + 1;
+                    int rate = 4 + (entityObj.getEntityId() % 7);
+
+                    int index1 = 0;
+
+                    HashMap<Integer, List<Integer>> lookupStateToSequence = new HashMap<>();
+
+                    List<Integer> listDelays = new ArrayList<>();
+
+                    listDelays.add(9);
+                    listDelays.add(3);
+                    listDelays.add(3);
+                    listDelays.add(3);
+                    listDelays.add(6);
+
+                    lookupStateToSequence.put(index1++, listDelays);
+                    lookupStateToSequence.put(index1++, listDelays);
+                    lookupStateToSequence.put(index1++, listDelays);
+
+                    listDelays = new ArrayList<>();
+                    listDelays.add(9);
```

Comment:
You could probably shorten this to List.addAll combined with Arrays.AsList. Something like: listDelays.addAll(Arrays.asList([9, 3, 3, 3, etc]));

Reply:
or `Lists.newArrayList(...)`